src/main/java/com/conveyal/r5/transit/TransitLayer.java

```diff
@@ -143,6 +144,16 @@
     /** Map from feed ID to feed CRC32 to ensure that we can't apply scenarios to the wrong feeds */
     public Map<String, Long> feedChecksums = new HashMap<>();
 
+    /**
+     * A string uniquely identifying the contents of this TransitLayer among all TransitLayers.
+     * When no scenario has been applied, this field will contain the ID of the containing TransportNetwork.
+     * When a scenario has modified this StreetLayer, this field will be changed to the scenario's ID.
+     * We need a way to know what transit information is in the layer independent of object identity, which is lost in
+     * a round trip through serialization. This also allows re-using cached information for multiple scenarios that
+     * don't modify the transit network. (This never happens yet, but will when we allow street editing.)
```

Comment:
Really though, we could preserve linkages any time a scenario creates no new stops, but it's probably not worth the effort.